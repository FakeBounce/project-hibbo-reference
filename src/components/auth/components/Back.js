import React from 'react';
import { TouchableOpacity } from 'react-native';
import { func, string, bool } from 'prop-types';

import back from 'assets/auth/nav/back.png';

import * as Styles from '../styles';

const Back = ({ goBack, title, isTos }) => {
  return (
    <Styles.SHeader isTos={isTos}>
      <TouchableOpacity
        activeOpacity={1}
        onPress={() => {
          goBack(title);
        }}
      >
        <Styles.SBackImage source={back} />
      </TouchableOpacity>
    </Styles.SHeader>
  );
};

Back.defaultProps = {
  title: '',
  isTos: false,
};

Back.propTypes = {
  goBack: func.isRequired,
  title: string,
  isTos: bool,
};

export default Back;
