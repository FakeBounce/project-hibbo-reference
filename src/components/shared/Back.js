import React from 'react';
import { Image } from 'react-native';
import { string, func, oneOfType, object, number } from 'prop-types';
import EStyleSheet from 'react-native-extended-stylesheet';

import TouchableRipple from 'shared/TouchableRipple';

import straightArrow from 'assets/shared/straightArrow.png';
import straightArrowBlack from 'assets/shared/straightArrowBlack.png';
import flipArrow from 'assets/shared/curvedArrow.png';
import flipWhiteArrow from 'assets/shared/curvedWhiteArrow.png';
import straightArrowWhite from 'assets/shared/straightArrowWhite.png';
import cross from 'assets/shared/cross.png';
import appStyles from 'styles/appStyles';

const styles = EStyleSheet.create({
  container: {
    padding: 13 + appStyles.size.cardMargin,
  },
  straight: {
    width: 30,
    maxHeight: 20,
  },
  cross: {
    width: 16,
    height: 16,
  },
  flip: {
    marginTop: -10,
    width: 41,
    height: 39,
  },
});

const selectImage = (type, color) => {
  switch (type) {
    case 'cross':
      return cross;
    case 'flip':
      if (color === 'white') return flipWhiteArrow;
      return flipArrow;
    case 'straight':
    default:
      if (color === 'black') {
        return straightArrowBlack;
      }
      if (color === 'white') {
        return straightArrowWhite;
      }
      return straightArrow;
  }
};

const selectStyle = type => {
  switch (type) {
    case 'cross':
      return styles.cross;
    case 'flip':
      return styles.flip;
    case 'straight':
    default:
      return styles.straight;
  }
};

const Back = ({ type, action, color, style }) => {
  return (
    <TouchableRipple
      style={[styles.container, style]}
      activeOpacity={1}
      onPress={() => {
        action();
      }}
    >
      <Image source={selectImage(type, color)} style={selectStyle(type)} />
    </TouchableRipple>
  );
};

Back.defaultProps = {
  color: 'yellow',
  style: {},
};

Back.propTypes = {
  type: string.isRequired,
  action: func.isRequired,
  color: string,
  style: oneOfType([number, string, object]),
};

export default Back;
