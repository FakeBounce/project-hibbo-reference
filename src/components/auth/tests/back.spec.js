import { TouchableOpacity } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Back from '../components/Back';

describe('Back', () => {
  it('should render the component', () => {
    const fn = jest.fn();
    const component = renderer.create(<Back goBack={fn} title="" />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should handle press events', () => {
    const fn = jest.fn();
    const tree = shallow(<Back goBack={fn} title="" />);

    tree.find(TouchableOpacity).at(0).simulate('press');
    expect(fn).toHaveBeenCalled();
  });
});
