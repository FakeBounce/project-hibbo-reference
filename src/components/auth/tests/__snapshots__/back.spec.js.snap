// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back should render the component 1`] = `
<View
  style={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={1}
      style={undefined}
    />
  </View>
</View>
`;
